scripts/sampling/taco_sampling.py

```diff
@@ -0,0 +1,98 @@
+import pprint
+import random
+from collections import defaultdict
+
+from nucypher.blockchain.eth import domains
+from nucypher.blockchain.eth.agents import (
+    ContractAgency,
+    TACoApplicationAgent,
+    TACoChildApplicationAgent,
+)
+from nucypher.blockchain.eth.registry import ContractRegistry
+
+domain = "mainnet"
+infura_id = "..."
+polygon_endpoint = "https://polygon-mainnet.infura.io/v3/" + infura_id
+eth_endpoint = "https://mainnet.infura.io/v3/" + infura_id
+dkg_size = 30
+# adopter_seed = 42
+# random.seed(adopter_seed)
+unlocking_delay = 182 * 24 * 60 * 60
+
+domain = domains.get_domain(domain)
+registry = ContractRegistry.from_latest_publication(domain=domain)
+
+child_application_agent = ContractAgency.get_agent(
+    agent_class=TACoChildApplicationAgent,
+    registry=registry,
+    blockchain_endpoint=polygon_endpoint,
+)
+
+taco_application_agent = ContractAgency.get_agent(
+    agent_class=TACoApplicationAgent,
+    registry=registry,
+    blockchain_endpoint=eth_endpoint,
+)
+
+# _, staking_providers_dict = child_application_agent.get_all_active_staking_providers()
+# staking_providers = list(staking_providers_dict.keys())
+
+
+def horizon(info: TACoApplicationAgent.StakingProviderInfo):
+    amount = info.authorized - info.deauthorizing
```

Comment:
I know this is draft and WIP 😅  , and of course perhaps there are other requirements to add in here, but in terms of authorized amounts logic, it may already be taken care of via the call to `getActiveStakingProviders()` 
 in `TACoApplication` - see https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/TACoApplication.sol#L699 (see call to `getEligibleAmount`).